src/SpectralDataManager.cpp

```diff
@@ -0,0 +1,214 @@
+#include <iostream>
+#include "FFT.h"
+#include "SpectralDataManager.h"
+#include "WaveTrack.h"
+
+SpectralDataManager::SpectralDataManager()= default;
+
+SpectralDataManager::~SpectralDataManager()= default;
+
+struct SpectralDataManager::Setting{
+   eWindowFunctions mInWindowType = eWinFuncHann;
+   eWindowFunctions mOutWindowType = eWinFuncHann;
+   size_t mWindowSize = 2048;
```

Comment:
I see the "magic number" 2048 repeated in this file.

Is the sameness of those numbers a coincidence, unimportant to the design -- or is it essential?  If it's essential, it is better to give this constant a name, and repeat the name in two places, not the literal number.

I think the hop size of 1/4 the window size may need to be used in more places outside this file -- so that hop size should be public in the header as well, as a named constant.